dark/reads.py

```diff
@@ -378,41 +414,62 @@ def propertyDetails(self):
         """
         return (PROPERTY_DETAILS.get(aa, NONE) for aa in self.sequence)
 
-    def ORFs(self):
+    def ORFs(self, openORFs=False):
         """
         Find all ORFs in our sequence.
 
         @return: A generator that yields AAReadORF instances that correspond
             to the ORFs found in the AA sequence.
         """
-        ORFStart = None
-        openLeft = True
-        seenStart = False
-
-        for index, residue in enumerate(self.sequence):
-            if residue == 'M':
-                # Start codon.
-                openLeft = False
-                seenStart = True
-            elif residue == '*':
-                # Stop codon. Yield an ORF, if it has non-zero length.
-                if ORFStart is not None and index - ORFStart > 0:
-                    # The ORF has non-zero length.
-                    yield AAReadORF(self, ORFStart, index, openLeft, False)
-                    ORFStart = None
-                # After a stop codon, we can no longer be open on the left
-                # and we have no longer seen a start codon.
-                openLeft = seenStart = False
-            else:
-                if (seenStart or openLeft) and ORFStart is None:
-                    ORFStart = index
 
-        # End of sequence. Yield the final ORF if there is one and it has
-        # non-zero length.
-        length = len(self.sequence)
-        if ((seenStart or openLeft) and ORFStart is not None and
-                length - ORFStart > 0):
-            yield AAReadORF(self, ORFStart, length, openLeft, True)
+        # Return open ORFs to the left and right and closed ORFs within the
+        # sequence.
+        if openORFs:
+            ORFStart = 0
+            inOpenORF = True  # open on the left
+            inORF = False
+
+            for index, residue in enumerate(self.sequence):
+                if residue == '*':
+                    if inOpenORF:
+                        if index:
+                            yield AAReadORF(self, ORFStart, index, True, False)
+                        inOpenORF = False
+                    if inORF:
+                        if ORFStart != index:
+                            yield AAReadORF(self, ORFStart, index,
+                                            False, False)
+                        inORF = False
+                if residue == 'M':
```

Comment:
This should definitely be an `elif`.  It's also faster because the interpreter just won't do this test if the residue was a `*`.

Reply:
done